towerlib/entities/job.py

```diff
@@ -793,8 +792,10 @@ def project(self):
             Inventory: The project that the job template is part of.
 
         """
-        url = self._data.get('related', {}).get('project')
-        return self._tower._get_object_by_url('Project', url)  # pylint: disable=protected-access
+        if self._data.get('related', {}).get('project'):
```

Comment:
Following the return early principle to be consistent please return with None if there is no value and return with the proper value on the exit.